### PR TITLE
run_pacta() gains arguments for IO

### LINKED ISSUE #26: Avoid jumping between bash and R
Right now we jump a lot bwtween bash and R. For example
we use ./bin/run-pacta and `R/run_pacta()`.

It would be nice to do all in one place.

I prefer to work in R mostly because that let us 1) test with
testthat and 2) develop interactively, both in normal and 
debugger mode.

@jdhoffa, I know you prefer to stay in the shell. Can you 
comment on why?

### PR DESCRIPTION
**Pull request by maurolepore**
Closes #26 

This PR moves all the work of `bin/run-pacta` to R via `run_pacta()`. 

Also:
* The input test files are not in inst/extdata. It's too confusing that out inputs live alongside the outputs.
* `run_pacta()` gains `input` and `output`, which now default to `../input` and `../output` but could be anything. The next step would be to pass those arguments via environmental variables we can get via `Sys.getenv()`.

### CODE REVIEW

File: R/run_pacta.R [hunk 1]
```diff
@@ -6,17 +6,33 @@
 #' @examples
 #' run_pacta()
 #' @noRd
-run_pacta <- function(wd = here::here()) {
+run_pacta <- function(wd = here::here(), input = "../input", output = "../output") {
```
Comment (maurolepore) [inline R/run_pacta.R:9]: TODO: Replace defaults to first consult environmental variables via `Sys.getenv()`, e.g.

```bash
# .env
PACTA_INPUT="../input"
PACTA_OUTPUT="../output"
...
```

```r
run_pacta <- function(wd = [...], input = Sys.getenv("PACTA_INPUT, unset = "../input", output = [...]) {
  [...]
}
```

This way the call becomes again simply `run_pacta()` (with no arguments) and we have a mechanism to pass input/ and output/ directories via environmental variables to `docker run` and `docker-compose`.



`run_pacta()` (with no arguments) and in Dockerfile

File: bin/run-pacta [hunk 2]
```diff
@@ -1,11 +1,8 @@
...
 parent=$(dirname $pacta)
 
-cp -r "$parent"/input/*.yml "$pacta"/working_dir/10_Parameter_File
-cp -r "$parent"/input/*.csv "$pacta"/working_dir/20_Raw_Inputs
-
-Rscript --vanilla -e "source('""$pacta""/R/run_pacta.R'); run_pacta(wd = '""$pacta""')"
-cp -r "$pacta"/working_dir "$parent"/output
+Rscript --vanilla -e \
+  "source('""$pacta""/R/run_pacta.R'); \
+   run_pacta(wd = '""$pacta""', input = '""$parent""/input', output = '""$parent""/output')"
```
Comment (maurolepore) [inline bin/run-pacta:8]: Once we use env varsiables the call to `Rscript` will be much simpler.

```bash
Rscript --vanilla -e "source('""$pacta""/R/run_pacta.R'); run_pacta()"
```

We might even call this directly from Dockerfile and remove bin/run-pacta entirely

File: bin/setup-io [hunk 3]
```diff
@@ -2,5 +2,5 @@
 
 mkdir -p ../input
 mkdir -p ../output
-cp working_dir/10_Parameter_File/TestPortfolio_Input_PortfolioParameters.yml ../input/
-cp working_dir/20_Raw_Inputs/TestPortfolio_Input.csv ../input/
+cp inst/extdata/TestPortfolio_Input_PortfolioParameters.yml ../input/
+cp inst/extdata/TestPortfolio_Input.csv ../input/
```
Comment (maurolepore) [inline bin/setup-io:6]: I think we could use inst/ quite heavily to store the crazy structure of the legacy code, and keep the root tidy.

**Review by maurolepore — COMMENTED** [review]
Some notes -- mostly to myself

**Review by jdhoffa — COMMENTED** [review]
I would prefer to wait and discuss my latest [comments](https://github.com/2DegreesInvesting/pactaCore/issues/26#issuecomment-898313193) on 26, before we merge this, if that's ok!

**Comment by jdhoffa** [comment]
Ok I see the point. You want IO in the function so you can run a regression test directly on `run_pacta.R`

**Comment by maurolepore** [comment]
> I would prefer to wait and discuss my latest comments on 26, before we merge this, if that's ok!

Sure! Also, I'm move to a draft as I still have some ideas I would like to push to clarify the benefit I see in this approach.

**Comment by maurolepore** [comment]
I advanced quite a bit but didn't make it yet to my final goal -- to run the tool from wherever, using environmental variables. I'll revisit this in a bit.

**Comment by maurolepore** [comment]
Closing in favor or a cleaner upcomming PR

**Comment by maurolepore** [comment]
Follow up at #42

**Pull request closed without merging**